### PR TITLE
Issue #10, create menu class 

### LINKED ISSUE #10: Create menu class
### Motivation

As a menu is central to any restaurant, implementing a menu class should be a priority. 
### Feature Description
The menu class will have one member variable, which will be a vector of items, representing all items on the menu.

It will have the following methods:

- `add_menu_item(item)`: adds a new MenuItem object to the menu_items list
- `remove_menu_item(item)`: removes a MenuItem object from the menu_items list
- `get_menu_item(item_name)`: returns the MenuItem object corresponding to the given item name
- `get_menu_item_price(item_name)`: returns the price of the MenuItem object corresponding to the given item name
- `print_menu()`: prints all the items on the menu

This is just a basic implementation of the menu class and more complexity can be added later.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Chaitanya8135**
adds Menu class as per #10 , assuming interface of Item class as in #6.
closes #10

### CODE REVIEW

File: src/Menu.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    } else {
+        return *it;
+    }
+}
+
+float Menu::get_menu_item_price(const std::string& item_name) {
+    std::vector<Item>::iterator it = find_item(item_name);
+    if (it == menu_items.end()) {
+        std::cout << "Item with Name: " << item_name << "doesn't exist !\n";
+        return -1.0;
```
Comment (RockyDesigne) [inline src/Menu.cpp:67]: should throw an error instead.

**Review by RockyDesigne — CHANGES_REQUESTED** [review]
Lacks error handling.

File: src/Menu.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Menu::~Menu() = default;
+
+std::vector<Item>::iterator Menu::find_item(int item_ID) {
+    std::vector<Item>::iterator it = menu_items.begin();
+    while (it != menu_items.end() && (*it).getItemID() != item_ID) {
+        it++;
+    }
+    return it;
+}
```
[the remark this answers is not in the document]
Comment (Chaitanya8135) [reply src/Menu.cpp:17]: Apologies, the `find_item` function was not meant to be exposed, it's now made private (can only be called within the class).
The exception mentioned is handled by the functions that call `find_item`.

File: include/sudoManager/Menu.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#ifndef SUDOMANAGER_MENU_HPP
+#define SUDOMANAGER_MENU_HPP
+
+#include<iostream>
+#include<vector>
+#include<stdexcept>
+#include "Item.hpp"
```
Comment (genciberisha) [inline include/sudoManager/Menu.hpp:7]:
```suggestion
#include "sudoManager/Item.hpp"
```

File: src/Menu.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#include "sudoManager/Menu.hpp"
+#include "Item.cpp"
```
Comment (genciberisha) [inline src/Menu.cpp:2]:
```suggestion
#include "sudoManager/Item.hpp"
```

**Review by genciberisha — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I have updated the base **cmake** system of the project, please pull from main
and resolve conflicts.

**Comment by genciberisha** [comment]
Pull from **main** before resolving the issues

**Pull request closed without merging**